tests/lib/api.js

```diff
@@ -41,4 +42,27 @@ describe("api", () => {
     it("should have SourceCode exposed", () => {
         assert.isFunction(api.SourceCode);
     });
+
+    describe("loadESLint", () => {
+        it("should be a function", () => {
+            assert.isFunction(api.loadESLint);
+        });
+
+        it("should return a Promise", () => {
+            assert.instanceOf(api.loadESLint(), Promise);
+        });
+
+        it("should return ESLint when useFlatConfig is true", async () => {
+            assert.strictEqual(await api.loadESLint({ useFlatConfig: true }), api.ESLint);
+        });
+
+        it("should return LegacyESLint when useFlatConfig is false", async () => {
+            assert.strictEqual(await api.loadESLint({ useFlatConfig: false }), LegacyESLint);
+        });
+
+        it("should return ESLint when useFlatConfig is not provided", async () => {
+            assert.strictEqual(await api.loadESLint(), api.ESLint);
+        });
```

Comment:
Can we add a test case for LegacyESLint here as well?

Reply:
Yup!